prusti-viper/src/encoder/pure_function_encoder.rs

```diff
@@ -1345,15 +1406,62 @@ impl<'p, 'v: 'p, 'tcx: 'v> BackwardMirInterpreter<'tcx>
                                 let array_types = self.encoder.encode_array_types(place_ty).with_span(span)?;
                                 state.substitute_value(&opt_lhs_value_place.unwrap(), array_types.array_len.into());
                             }
-                            _ => {
-                                return Err(SpannedEncodingError::unsupported(
-                                    "checking the length of anything but arrays in pure code is not implemented yet",
-                                    span,
-                                ));
+                            ty::TyKind::Slice(..) => {
+                                let snap_len = self.encoder.encode_snapshot_slice_len(
+                                    place_ty,
+                                    self.encode_place(place).with_span(span)?.0,
+                                ).with_span(span)?;
+
+                                state.substitute_value(&opt_lhs_value_place.unwrap(), snap_len);
                             }
+                            _ => span_bug!(span, "length should only be requested on arrays or slices"),
```

Comment:
Does a `span_bug!` turn into a panic? If so, can we use an `internal` error instead?

Reply:
i think it does, just like `unreachable!()` which we thought about earlier. what made me think more towards `span_bug!` instead of an internal error is that as far as we understand it, if we encounter anything else than array or slice here, it's a bug in `rustc` more than in prusti.

Reply:
Ah, you're right, it's the `Len` thing. Never mind then, `span_bug` to panic makes sense.